crates/host/src/wasmbus/mod.rs

```diff
@@ -404,42 +405,108 @@ pub struct Host {
 /// Given the NATS address, authentication jwt, seed, tls requirement and optional request timeout,
 /// attempt to establish connection.
 ///
+/// This function should be used to create a NATS client for Host communication, for non-host NATS
+/// clients we recommend using async-nats directly.
 ///
 /// # Errors
 ///
 /// Returns an error if:
 /// - Only one of JWT or seed is specified, as we cannot authenticate with only one of them
 /// - Connection fails
-async fn connect_nats(
+pub async fn connect_nats(
     addr: impl async_nats::ToServerAddrs,
     jwt: Option<&String>,
     key: Option<Arc<KeyPair>>,
     require_tls: bool,
     request_timeout: Option<Duration>,
+    workload_identity_config: Option<WorkloadIdentityConfig>,
 ) -> anyhow::Result<async_nats::Client> {
-    let opts = async_nats::ConnectOptions::new().require_tls(require_tls);
-    let opts = match (jwt, key) {
-        (Some(jwt), Some(key)) => opts.jwt(jwt.to_string(), {
-            move |nonce| {
+    let opts = match (jwt, key, workload_identity_config) {
+        (jwt, key, Some(wid_cfg)) => {
```

Comment:
I think that you may want to move the match block where you check for only one of jwt/seed above this to ensure this doesn't match first. I doubt you'd reach this point anyways so we can just address that in a follow up

Reply:
I'm making a note to fix this in post to keep moving this forward, I hope you're ok with that 🙂